src/Libraries/Microsoft.Extensions.AI.Evaluation.NLP/Common/MatchCounter.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI.Evaluation.NLP.Common;
+
+[DebuggerDisplay("{ToDebugString(),nq}")]
+internal readonly struct MatchCounter<T> : IEnumerable<KeyValuePair<T, int>>
+    where T : IEquatable<T>
+{
+    private readonly Dictionary<T, int> _counts = [];
+
+    public readonly int Sum => _counts.Values.Sum();
+
+    public MatchCounter()
```

Comment:
Looks like this may have been missed from the previous round - but it would be nice to remove the empty default constructor by making `items` optional (and nullable) on the below overload.

Reply:
Sure, there is no functional difference - but it would probably be fewer lines of code (and would avoid the need to throw on null below etc.)